docs/concepts/index.md

```diff
@@ -0,0 +1,160 @@
+
+        def parse(data):
+            if "y" not in data["format_errors"] and data["submitted_answers"]["y"] < 0:
+                data["format_errors"]["y"] = "Negative numbers are not allowed"
+
+        def grade(data):
+            if math.isclose(data["score"], 0.0) and data["submitted_answers"]["y"] > data["params"]["x"]:
+                data["partial_scores"]["y"]["score"] = 0.5
+                data["score"] = 0.5
+                data["feedback"]["y"] = "Your value for $y$ is larger than $x$, but incorrect."
+        ```
+
+#### External Graders
+
+For coding questions, you can use an [**external grader**](../externalGrading.md) to grade a submission instead of a `server.py` file. This lets you grade student-submitted code in a sandboxed environment. There are currently external graders for C, C++, Java, Python, and more.
+
+#### Workspaces
+
+Additionally, it may be useful for students to work in a consistent development [**workspace**](../workspaces/index.md) to reduce the likelihood of issues with their environment, and provide a code editor with syntax highlighting, autocompletion, and other features. Workspaces provide a seamless coding experience for students when used with an external autograder. Workspaces use [Docker](https://docker.com) to ensure a consistent environment, with premade workspace for popular languages (e.g. Python, C++). Workspaces based on custom Docker images are also supported.
+
+![workspace screenshot](./workspace.png)
+Both of these features can be setup by modifying the `info.json` file for the question.
+
+### Elements
+
+Questions can use [**elements**](../elements.md) to accept student input, display diagrams, control how questions are displayed. Elements are pre-built interface components for common questions types, including numerical inputs, multiple choice, and more. If you require more flexibility, [**custom elements**](../devElements.md) are reusable components that can be tailored to your course to create more complex questions.
+
+=== "Integer input element"
+
+    ![Integer input element](../elements/pl-integer-input.png)
+
+=== "HTML for element"
+
+    ```html title="question.html"
+    <pl-integer-input answers-name="int_value" label="$y =$"></pl-integer-input>
+    ```
+
+You can view a list of all the available elements in the [elements documentation](../elements.md).
+
+### Assessments
+
+A course contains a collection of questions, which are composed together to create [**assessments**](../assessment/index.md). Each course instance contains a collection of **homework assessments** and **exam assessments**. Homeworks are formative assessments designed for mastery learning: they give unlimited retries with different variants, and incentivize repeated correct answers. Exams are summative assessments designed to measure learning: they give limited attempts with the same variant, and incentivize getting the correct answer on the first try.
```

Comment:
Nit, is this any better?

```suggestion
A course's questions are composed together to create homework and exam [**assessments**](../assessment/index.md). Homeworks are formative assessments designed for mastery learning: they give unlimited retries with different variants, and incentivize repeated correct answers. Exams are summative assessments designed to measure learning: they give limited attempts with the same variant, and incentivize getting the correct answer on the first try.
```

Reply:
Hmm, this removes the point that assessments belong to a course instance

Reply:
This?

```suggestion
A course's questions are composed together to create homework and exam [**assessments**](../assessment/index.md) in a course instance. Homeworks are formative assessments designed for mastery learning: they give unlimited retries with different variants, and incentivize repeated correct answers. Exams are summative assessments designed to measure learning: they give limited attempts with the same variant, and incentivize getting the correct answer on the first try.
```